src/frontend/file_chooser.rs

```diff
@@ -100,6 +99,7 @@ impl SimpleComponent for FileChooser {
         sender: relm4::prelude::ComponentSender<Self>,
     ) -> relm4::prelude::ComponentParts<Self> {
         if is_wayland() {
+            root.set_keyboard_mode(gtk4_layer_shell::KeyboardMode::OnDemand);
             root.init_layer_shell();
             root.set_layer(Layer::Overlay);
```

Comment:
```suggestion
            root.init_layer_shell();
            root.set_layer(Layer::Overlay);
            root.set_keyboard_mode(gtk4_layer_shell::KeyboardMode::OnDemand);
```

Otherwise we get a warning when calling `set_keyboard_mode` as our window is not a layer surface yet.